kvbc/src/categorization/kv_blockchain.cpp

```diff
@@ -143,101 +143,63 @@ std::future<BlockDigest> KeyValueBlockchain::computeParentBlockDigest(const Bloc
-        (void)values;
-        // cat.multiGet(keys,versions,values);
-        return;
-      },
-      category);
+  std::visit([&keys, &versions, &values](const auto& category) { category.multiGet(keys, versions, values); },
+             getCategory(category_id));
 }
 
-void KeyValueBlockchain::multiGetLatest(const std::string& cat_id,
+void KeyValueBlockchain::multiGetLatest(const std::string& category_id,
                                         const std::vector<std::string>& keys,
                                         std::vector<std::optional<Value>>& values) const {
-  auto category = getCategory(cat_id);
-  std::visit(
-      [&keys, &values](const auto& cat) {
-        (void)keys;
-        (void)values;
-        // cat.multiGetLatest(keys,values);
-        return;
-      },
-      category);
+  std::visit([&keys, &values](const auto& category) { category.multiGetLatest(keys, values); },
+             getCategory(category_id));
 }
 
-std::optional<BlockId> KeyValueBlockchain::getLatestVersion(const std::string& cat_id, const std::string& key) const {
-  auto category = getCategory(cat_id);
-  std::visit(
-      [&key](const auto& cat) {
-        (void)key;
-        // cat.getLatestVersion(key);
-        return;
-      },
-      category);
+std::optional<categorization::TaggedVersion> KeyValueBlockchain::getLatestVersion(const std::string& category_id,
+                                                                                  const std::string& key) const {
+  std::optional<categorization::TaggedVersion> ret;
+  std::visit([&key, &ret](const auto& category) { ret = category.getLatestVersion(key); }, getCategory(category_id));
+  return ret;
+}
 
-  return std::optional<BlockId>{};
+void KeyValueBlockchain::multiGetLatestVersion(
+    const std::string& category_id,
+    const std::vector<std::string>& keys,
+    std::vector<std::optional<categorization::TaggedVersion>>& versions) const {
+  std::visit([&keys, &versions](const auto& catagory) { catagory.multiGetLatestVersion(keys, versions); },
+             getCategory(category_id));
 }
 
-void KeyValueBlockchain::multiGetLatestVersion(const std::string& cat_id,
-                                               const std::vector<std::string>& keys,
-                                               std::vector<std::optional<BlockId>>& versions) const {
-  auto category = getCategory(cat_id);
-  std::visit(
-      [&keys, &versions](const auto& cat) {
-        (void)keys;
-        (void)versions;
-        // cat.multiGetLatestVersion(keys,values);
-        return;
-      },
-      category);
+Updates KeyValueBlockchain::getBlockUpdates(BlockId block_id) const {
+  return Updates{std::move(getRawBlock(block_id).data.updates)};
 }
 
-CategoryInput KeyValueBlockchain::getBlockData(BlockId block_id) { return getRawBlock(block_id).data.updates; }
+CategoryInput KeyValueBlockchain::getCategoryInput(const Updates& updates) const { return updates.category_updates_; }
```

Comment:
Do we essentially need this method to use **CategoryInput**'s operator==() from CMF? If yes, for convenience, we might define operator==() for `Updates` too.